src/NVMLIOGroup.cpp

```diff
@@ -0,0 +1,673 @@
+                              {"NVML::THROTTLE_REASONS", {
+                                  "Accelerator clock throttling reasons",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::average,
+                                  string_format_double
+                                  }},
+                              {"NVML::TEMPERATURE", {
+                                  "Accelerator temperature in degrees Celsius",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::average,
+                                  string_format_double
+                                  }},
+                              {"NVML::TOTAL_ENERGY_CONSUMPTION", {
+                                  "Accelerator energy consumption in joules since the driver was loaded",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::sum,
+                                  string_format_double
+                                  }},
+                              {"NVML::PERFORMANCE_STATE", {
+                                  "Accelerator performance state",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::average,
+                                  string_format_double
+                                  }},
+                              {"NVML::PCIE_RX_THROUGHPUT", {
+                                  "Accelerator PCIE receive throughput in bytes per second over a 20 millisecond period",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::average,
+                                  string_format_double
+                                  }},
+                              {"NVML::PCIE_TX_THROUGHPUT", {
+                                  "Accelerator PCIE transmit throughput in bytes per second over a 20 millisecond period",
+                                  {},
+                                  GEOPM_DOMAIN_BOARD_ACCELERATOR,
+                                  Agg::average,
+                                  string_format_double
+                                  }},
+                              {"NVML::CPU_ACCELERATOR_ACTIVE_AFFINITIZATION", {
+                                  "Returns the associated accelerator for a given CPU as determined by running processes."
+                                  "\n  If no accelerators map to the CPU then NAN is returned"
+                                  "\n  If multiple accelerators map to the CPU -1 is returned",
+                                  {},
+                                  GEOPM_DOMAIN_CPU,
+                                  Agg::average,
```

Comment:
expect same

Reply:
NAN is used to represent a CPU not being affinitized to an accelerator when reading at the CPU level.  -1 is used to indicate a CPU is associated with multiple GPUs when reading at the CPU level.  If we read a CORE with two CPUs that are affinitized to two different GPUs we will return NAN with expect_same and not -1.  

The cleanest solution might be to change to expect_same and modify the returns to NAN --> affinitization with multiple GPUs and -1 --> no affinitization.

Reply:
I've swapped the returns on CPU_ACCELERATOR_AFFINITZATION such that -1 means not affinitized, NAN means affinitized to multiple accelerators.  This pulls per CPU usage in line with aggregation usage as outlined above.